### PR TITLE
Separate build tests to its own pipeline

### PR DESCRIPTION
**Pull request by jashook**
This should not be merged without a separate definition change to the
official build pipeline.

/cc @RussKeldorph

### CODE REVIEW

**Comment by jashook** [comment]
@dotnet-bot help

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by jashook** [comment]
@mmitche this is ready to review. Note this does not remove the definition from the original pipelines.json yet. It will run tests twice to make sure the infrastructure works on the first couple of runs.

**Comment by weshaggard** [comment]
What is the goal of creating a completely different pipeline?

**Comment by jashook** [comment]
It allows us to conditionally wait on the test pipeline. So for prodcon we can launch the testing but not block the rest of the upstack components.

**Comment by weshaggard** [comment]
@jashook @mmitche so is the plan to add this extra test pipeline to the orchestrated build pipeline but doing it in parallel with the repo's above coreclr?

File: buildpipeline/tests/Dotnet-CoreClr-Trusted-BuildTests.json [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+      "condition": "ne(variables['PB_SkipTests'], 'true')",
+      "refName": "Task3",
+      "task": {
+        "id": "d9bafed4-0b18-4f58-968d-86655b4d2ce9",
+        "versionSpec": "1.*",
+        "definitionType": "task"
+      },
+      "inputs": {
+        "filename": "sync.cmd",
+        "arguments": "-ab -AzureAccount=$(CloudDropAccountName) -AzureToken=$(CloudDropAccessToken) -Container=$(ParentContainerName) -RuntimeId=$(Rid) -BlobNamePrefix=$(ParentBlobNamePrefix)$(ParentBuildType)/TestNativeBins/$(Rid) -- /p:DownloadDirectory=$(Build.SourcesDirectory)/packages/TestNativeBins/$(Rid)/$(ParentBuildType)",
```
Comment (weshaggard) [inline buildpipeline/tests/Dotnet-CoreClr-Trusted-BuildTests.json]: Given this is an independent pipeline what is the strategy for finding the right azure blob for the matching coreclr bits?

Comment (mmitche) [reply buildpipeline/tests/Dotnet-CoreClr-Trusted-BuildTests.json]: The blob is statically predictable given the input parameters to the build pipeline and the build pipeline's build id.

**Comment by mmitche** [comment]
> @jashook @mmitche so is the plan to add this extra test pipeline to the orchestrated build pipeline but doing it in parallel with the repo's above coreclr?

@weshaggard Yes

**Comment by jashook** [comment]
ping @weshaggard @mmitche 
/cc @RussKeldorph

File: buildpipeline/tests/Dotnet-CoreClr-Trusted-BuildTests.json (hunk 1 above)
Comment (weshaggard) [reply buildpipeline/tests/Dotnet-CoreClr-Trusted-BuildTests.json]: OK. That does mean that each of the these pipelines will need to be kicked off with same input parameters. Do we have any good way to ensure that?

**Pull request merged**